src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1075,24 +1077,62 @@ public void failActivityTask(String activityId, RespondActivityTaskFailedRequest
       throws InternalServiceError, EntityNotExistsError, BadRequestError {
     update(
         ctx -> {
-          StateMachine<?> activity = getActivity(activityId);
+          log.info("Fail activity time=" + store.currentTimeMillis());
+
+          StateMachine<ActivityTaskData> activity = getActivity(activityId);
           activity.action(StateMachines.Action.FAIL, ctx, request, 0);
-          activities.remove(activityId);
-          ctx.unlockTimer();
-          scheduleDecision(ctx);
+          if (isTerminalState(activity.getState())) {
+            activities.remove(activityId);
+            ctx.unlockTimer();
+            scheduleDecision(ctx);
+          } else {
+            addActivityRetryTimer(ctx, activity);
```

Comment:
Time skipping seems not working inside retry window.

Reply:
Fixed

Reply:
Fixed